### PR TITLE
pipe-shell evolutions (#3356 + conflict resolutions)

### PR DESCRIPTION
**Pull request by mmandlis**
patched https://github.com/PolymerLabs/arcs/pull/3356/ and resolved conflicts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,6 +86,7 @@
     "wdio-chromedriver-service": "^5.0.2",
     "webdriverio": "^5.11.1",
     "webpack": "^4.32.2",
+    "webpack-auto-inject-version": "^1.2.2",
```
Comment (shaper) [inline package.json]: Do we really need this as part of the PR? Ask @sjmiles or try removing?

File: shells/pipes-shell/README.md [hunk 2]
```diff
@@ -1,90 +1,43 @@
...
 
-- pipes-shell
-  * // platform-agnostic sources
-- node/
-  - serve.sh // entry-point to execute the NodeJs application
-  ...
-- web/
-  - index.html // entry-point to execute the Browser application
-  ...
+Plaform code is in the `web` or `node` folders.
```
Comment (shaper) [inline shells/pipes-shell/README.md]: Plaform -> Platform

File: package.json (hunk 1 above)
Comment (mmandlis) [reply package.json]: i trying removing it, and all seems to work fine without it.

File: shells/pipes-shell/README.md (hunk 2 above)
Comment (mmandlis) [reply shells/pipes-shell/README.md]: fixed

**Comment by mmandlis** [comment]
any objections / concerns for me merging this?

**Comment by shaper** [comment]
I support merging it. For next import @yuangu-google will probably need to update the Android side where I suspect we are referencing pipe-shell-2 today but he needs to do that in any case given the changes we're making.

**Pull request merged**